src/components/NestedForm/NestedFormBuilder.tsx

```diff
@@ -38,6 +36,21 @@ export class NestedFormBuilder extends React.PureComponent<data.IFieldBuilderPro
         return true;
     }
 
+        // NOTE(andrews) canDrop holds the business logic for determining if the source item can
+    // be dropped on the target item.
+    private canDrop(source: data.IDragSourceItem, target: data.IDropTargetItem): boolean {
+        // NOTE(andrews): Prohibit more than one layer of nested fields
+        if (source.field.type === 'NestedForm' &&  target.parentId) {
```

Comment:
source.field.type === this.prop.field.type. Is this better instead of hard coding?